### PR TITLE
chore: BalCoW v1.0.0 release

### PR DESCRIPTION
**Pull request by wei3erHase**
PR addresses changes proposed by Wonderland's Internal Review [W] and Certora's Security Assessment [C]. 
- [x] [W,C] using OZ SafeTransfer #115
- [x] [W,C] rm `EMPTY_COMMITMENT` (unused) #115
- [x] [W] declare `MAX_ORDER_DURATION` in `minutes` #115
- [x] [W] share lock mechanism between reentrancy guard and commitment #115 
- [x] [C] renaming errors #115
- [x] [C] adding zero address checks on setters #132
- [x] [C] addressing gas optimizations #131 
- [x] [W] fixing typos and style consistency #133 
- [x] [W] make finalized check a modifier #134 
- [ ] [W] ~~add address argument to `TokenNotBound` error #136~~
- [x] [C] fixing incorrect error msgs #138 
- [x] [C] fixing `MAX_ORDER_DURATION` check #138 
- [x] [W] make the pool factory public #143 

**extras**
- [x] changing `bLabs` naming for `bDao` #147 
- [x] increasing max swap fee to 99% #158 
- [x] re-do unit tests with BTT approach
- [ ] advanced testing (echidna and halmos)

### CODE REVIEW

File: src/contracts/BPool.sol [hunk 1]
```diff
@@ -505,13 +491,16 @@ contract BPool is BToken, BMath, IBPool {
...
+
+    spotPrice = calcSpotPrice(
       IERC20(tokenIn).balanceOf(address(this)),
       inRecord.denorm,
       IERC20(tokenOut).balanceOf(address(this)),
       outRecord.denorm,
       _swapFee
     );
+
+    return spotPrice;
```
Comment (turtlemoji) [inline src/contracts/BPool.sol:503]: needed?

File: src/contracts/BPool.sol [hunk 2]
```diff
@@ -524,13 +513,16 @@ contract BPool is BToken, BMath, IBPool {
...
+
+    spotPrice = calcSpotPrice(
       IERC20(tokenIn).balanceOf(address(this)),
       inRecord.denorm,
       IERC20(tokenOut).balanceOf(address(this)),
       outRecord.denorm,
       0
     );
+
+    return spotPrice;
```
Comment (turtlemoji) [inline src/contracts/BPool.sol:525]: needed?

File: src/contracts/BPool.sol [hunk 3]
```diff
@@ -660,35 +643,35 @@ contract BPool is BToken, BMath, IBPool {
...
+  function _mintPoolShare(uint256 amount) internal virtual {
     _mint(address(this), amount);
   }
 
   /**
    * @dev Burns an amount of pool tokens.
    * @param amount The amount of pool tokens to burn
    */
-  function _burnPoolShare(uint256 amount) internal {
+  function _burnPoolShare(uint256 amount) internal virtual {
```
Comment (turtlemoji) [inline src/contracts/BPool.sol:662]: why the change?

**Review by turtlemoji — APPROVED** [review]
approved!

File: src/contracts/BToken.sol [hunk 4]
```diff
@@ -12,46 +12,46 @@ contract BToken is ERC20 {
...
-   * @return True if the operation is successful.
+   * @param spender The address which will spend the funds.
+   * @param amount The amount of tokens to increase the allowance by.
+   * @return success True if the operation is successful.
    */
-  function increaseApproval(address dst, uint256 amt) external returns (bool) {
-    _approve(msg.sender, dst, allowance(msg.sender, dst) + amt);
+  function increaseApproval(address spender, uint256 amount) external returns (bool success) {
+    _approve(msg.sender, spender, allowance(msg.sender, spender) + amount);
     return true;
```
Comment (turtlemoji) [inline src/contracts/BToken.sol:21]: We are doing different types of returns (sometimes by value, others no return, etc)

File: src/contracts/BToken.sol [hunk 5]
```diff
@@ -12,46 +12,46 @@ contract BToken is ERC20 {
...
-      _approve(msg.sender, dst, 0);
+  function decreaseApproval(address spender, uint256 amount) external returns (bool success) {
+    uint256 oldValue = allowance(msg.sender, spender);
+    if (amount > oldValue) {
+      _approve(msg.sender, spender, 0);
     } else {
-      _approve(msg.sender, dst, oldValue - amt);
+      _approve(msg.sender, spender, oldValue - amount);
     }
     return true;
```
Comment (turtlemoji) [inline src/contracts/BToken.sol:37]: idem here

File: src/contracts/BPool.sol (hunk 1 above)
Comment (wei3erHase) [reply src/contracts/BPool.sol:503]: is not needed, yet it was chosen to maintain consistency across methods, most of them already explicitly returned the calculated variable.

we went for:
- no naming return var in very simple returns (e.g. the `return true` in approvals)
- naming return var and explicitly returning it (to maintain code clarity)

this came handy for example in the Helper, that complying to an interface forced us to include a return var that was unused, then we could have not named it (with solidity warnings) or name it and return it empty (no warnings)

File: src/contracts/BPool.sol (hunk 3 above)
Comment (wei3erHase) [reply src/contracts/BPool.sol:662]: @0xteddybear ?
i believe is in order to be able to mock these methods useful for BTT approach

File: src/contracts/BPool.sol (hunk 2 above)
Comment (wei3erHase) [reply src/contracts/BPool.sol:525]: answered above

File: src/contracts/BPool.sol (hunk 3 above)
Comment (0xteddybear) [reply src/contracts/BPool.sol:646]: precisely! if the method is virtual, `foundry-smock` will generate methods to call this skipping all checks, and promoting it to an external call, allowing us to assert is has been called.

**Review by turtlemoji — DISMISSED** [review]

File: test/integration/BPool.t.sol [hunk 6]
```diff
@@ -53,19 +58,27 @@ abstract contract PoolSwapIntegrationTest is Test, GasSnapshot {
...
+    deal(address(weth), joiner.addr, WETH_LP_AMOUNT);
+    deal(address(wbtc), joiner.addr, WBTC_LP_AMOUNT);
+    deal(address(pool), exiter.addr, ONE_UNIT, false);
 
     dai.approve(address(pool), type(uint256).max);
     weth.approve(address(pool), type(uint256).max);
+    wbtc.approve(address(pool), type(uint256).max);
     pool.bind(address(dai), DAI_LP_AMOUNT, 8e18); // 80% weight
     pool.bind(address(weth), WETH_LP_AMOUNT, 2e18); // 20% weight
+    pool.bind(address(wbtc), WBTC_LP_AMOUNT, 10e18); // +100% weight (unused in swaps)
```
Comment (0xteddybear) [inline test/integration/BPool.t.sol:81]: now it's 40% 10% 10%

File: test/unit/BCoWPool/BCoWPoolBase.sol
Comment (0xteddybear) [inline test/unit/BCoWPool/BCoWPoolBase.sol:1]: we didn't remove this :sob:

File: test/unit/BCoWPool/BCoWPoolBase.t.sol [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+import {ISettlement} from 'interfaces/ISettlement.sol';
+import {MockBCoWPool} from 'test/manual-smock/MockBCoWPool.sol';
+
+contract BCoWPoolBase is BPoolBase, BCoWConst, BNum {
+  bytes32 public appData = bytes32('appData');
+  address public cowSolutionSettler = makeAddr('cowSolutionSettler');
+  bytes32 public domainSeparator = bytes32(bytes2(0xf00b));
+  address public vaultRelayer = makeAddr('vaultRelayer');
+  address public tokenIn;
```
Comment (0xteddybear) [inline test/unit/BCoWPool/BCoWPoolBase.t.sol:16]: we could get rid of these to be consistent with `BPoolBase`

File: test/unit/BPool/BPool_Bind.t.sol [hunk 8]
```diff
@@ -0,0 +1,108 @@
...
+
+contract BPoolBind is BPoolBase {
+  uint256 public tokenBindBalance = 100e18;
+  uint256 public tokenWeight = 1e18;
+  uint256 public totalWeight = 10e18;
+
+  function setUp() public virtual override {
+    super.setUp();
+
+    vm.mockCall(tokens[0], abi.encodePacked(IERC20.transferFrom.selector), abi.encode());
```
Comment (0xteddybear) [inline test/unit/BPool/BPool_Bind.t.sol:16]: opinion on giving `tokens[0]` its own variable?

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+  uint256 public tokenOutBalance = 400e18;
+  uint256 public totalWeight = 9e18;
+  uint256 public poolAmountIn = 1e18;
+  // calcSingleOutGivenPoolIn(400, 3, 100, 9, 1, 10^(-6))
+  uint256 public expectedAmountOut = 11.880392079733333329e18;
+
+  function setUp() public virtual override {
+    super.setUp();
+    tokenOut = tokens[1];
+    bPool.set__records(tokenOut, IBPool.Record({bound: true, index: 0, denorm: tokenOutWeight}));
```
Comment (0xteddybear) [inline test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol:25]: ...we don't use it for anything, but index is wrong

File: test/unit/BMath.tree [hunk 10]
```diff
@@ -0,0 +1,143 @@
...
+        └── ((((pS + ao) / pS) ^ (wT/wi))*bi - bi) / (1 - ((1 - (wi/wT)) * sf))
+
+BMathTest::calcSingleOutGivenPoolIn
+├── when pool supply is zero
+│   └── it should revert // division by zero
+├── when total weight is zero
+│   └── it should revert // division by zero
+├── when token balance out is zero
+│   └── it should return zero
+├── when swap fee and exit fee are zero
```
Comment (0xteddybear) [inline test/unit/BMath.tree:108]: in retrospect, do you consider all of these testcases would have been easier/cleaner with an array of valid scenarios, like I just did for `bPow`?

**Review by 0xteddybear — COMMENTED** [review]
most comments are questions, a few can be easily fixed, can take care of those tomorrow

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol (hunk 9 above)
Comment (wei3erHase) [reply test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol:25]: is because is not being used, set to default..

File: test/integration/BPool.t.sol (hunk 6 above)
Comment (wei3erHase) [reply test/integration/BPool.t.sol:81]: nee, is indistinct, because we're never swapping WBTC, from DAI-WETH perspective pool is `80-20`

File: test/unit/BMath.tree (hunk 10 above)
Comment (wei3erHase) [reply test/unit/BMath.tree:108]: wdym? we require both to be zero to revert 🤔

Comment (0xteddybear) [reply test/unit/BMath.tree:108]: I commented on the wrong line, sorry. I meant all of the non-reverting cases that `should return correct value`

**Review by 0xteddybear — APPROVED** [review]

File: package.json [hunk 11]
```diff
@@ -1,30 +1,35 @@
...
-    "url": "https://github.com/balancer-labs/balancer-core/issues"
+    "url": "https://github.com/balancer/cow-amm/issues"
   },
   "repository": {
     "type": "git",
-    "url": "git+https://github.com/balancer-labs/balancer-core.git"
+    "url": "git+https://github.com/balancer/cow-amm.git"
   },
-  "license": "GPL-3.0-only",
+  "license": "GPL-3.0-or-later",
```
Comment (turtlemoji) [inline package.json:14]: why or later?

**Review by turtlemoji — APPROVED** [review]
